creusot-contracts/src/std/vec.rs

```diff
@@ -75,6 +75,12 @@ extern_spec! {
     fn std::vec::Vec::push<T, A : Allocator>(self_: &mut Vec<T, A>, v: T)
 }
 
+extern_spec! {
+    #[ensures(result == None ==> (@*self_).len() == 0 && *self_ == ^self_)]
+    #[ensures(forall<r: T> result == Some(r) ==> (@*self_) == (@^self_).push(r))]
```

Comment:
I merged a version of this in #384, which used a match as the quantifier based version wasn't sufficient for the solvers. I really wish that the support for ADTs wasn't so partial...

Reply:
Yeah, I've seen this. I'm currently merging.